test/python/transpiler/test_adjacent_barriers.py

```diff
@@ -217,6 +217,62 @@ def test_barriers_with_blocking_obstacle_long(self):
 
         self.assertEqual(result, circuit_to_dag(expected))
 
+    def test_barriers_with_blocking_obstacle_narrow(self):
+        """ Test that barriers don't merge if there is an obstacle that
+            is blocking
+                 ░ ┌───┐ ░                     ░ ┌───┐ ░
+        q_0: |0>─░─┤ H ├─░─           q_0: |0>─░─┤ H ├─░─
+                 ░ └───┘ ░     ->              ░ └───┘ ░
+        q_1: |0>─░───────░─           q_1: |0>─░───────░─
+                 ░       ░                     ░       ░
+        """
+        qr = QuantumRegister(2, 'q')
+
+        circuit = QuantumCircuit(qr)
+        circuit.barrier(qr)
+        circuit.h(qr[0])
+        circuit.barrier(qr)
+
+        expected = QuantumCircuit(qr)
+        expected.barrier(qr)
+        expected.h(qr[0])
+        expected.barrier(qr)
+
+        pass_ = MergeAdjacentBarriers()
+        result = pass_.run(circuit_to_dag(circuit))
+
+        self.assertEqual(result, circuit_to_dag(expected))
+
+    def test_barriers_with_blocking_obstacle_twoQ(self):
+        """ Test that barriers don't merge if there is an obstacle that
+            is blocking
+
+                 ░       ░                     ░       ░
+        q_0: |0>─░───────░─           q_0: |0>─░───────░─
+                 ░       ░                     ░       ░
+        q_1: |0>─░───■─────    ->     q_1: |0>─░───■─────
+                 ░ ┌─┴─┐ ░                     ░ ┌─┴─┐ ░
+        q_2: |0>───┤ X ├─░─           q_2: |0>───┤ X ├─░─
+                   └───┘ ░                       └───┘ ░
```

Comment:
for this test I'm not sure why the top-right barrier would not merge with the left barrier. What's blocking it?

Reply:
Limitation of the drawer. There are only two barriers, left (on q0, q1) and right (q0, q2). Even though the barriers are adjacent (on q0) they can't be merged due to the dependency through the CX on q1,q2.

Reply:
gotcha.